### PR TITLE
2617 array params

### PR DESCRIPTION
**Pull request by dimovnike**
Pull request for #2671 but also a support for ObjectId was added in CLI. However the last needs strfmt update to work properly.

### CODE REVIEW

File: generator/templates/cli/registerflag.gotmpl [hunk 1]
```diff
@@ -66,6 +66,24 @@ if err := cmd.RegisterFlagCompletionFunc({{ camelize .Name }}FlagName,
     {{- end }}
 {{ end }}
 
+{{ define "arrayregistrator" }}
+    {{- if or (eq .GoType "[]int64") (eq .GoType "[]int32") (eq .GoType "[]string") (eq .GoType "[]float64") (eq .GoType "[]float32") (eq .GoType "[]bool") }}
+        {{ template "flagdescriptionvar" . }}
+        {{ template "flagnamevar" . }}
+        {{ template "flagdefaultvar" . }}
+        _ = cmd.PersistentFlags().{{ pascalize .GoType }}Slice({{ camelize .Name }}FlagName, {{ camelize .Name }}FlagDefault, {{ (camelize .Name) }}Description)
+        {{ template "enumcompletion" . }}
```
Comment (youyuanwu) [inline generator/templates/cli/registerflag.gotmpl:75]: "enumcompletion" does not support slice type. Maybe you can try 
```
{{ template "enumcompletion" .Item }} 
```
To complete array element

File: generator/templates/cli/retrieveflag.gotmpl [hunk 2]
```diff
@@ -22,6 +22,17 @@
...
+        {{ $flagValueVar }}Str, err := cmd.Flags().GetString({{ $flagNameVar }})
+        if err != nil{
+            return err, false
+        }
+        {{ $flagValueVar }} := strfmt.ObjectId{}
+		if err := {{ $flagValueVar }}.UnmarshalText([]byte({{ $flagValueVar }}Str)); err != nil{
+            return err, false
+        }
+        m.{{ pascalize .Name }} = {{- if .IsNullable }}&{{- end }}{{ $flagValueVar }}
     {{- else if eq .GoType "strfmt.UUID" }} {{/*Many of the strfmt string alias types can be added here*/}}
```
Comment (youyuanwu) [inline generator/templates/cli/retrieveflag.gotmpl:36]: Maybe 
```
{{- else if or (eq .GoType "strfmt.UUID") (eq .GoType "strfmt.ObjectId") }}
```
Can work without copy paste the majority of the texts

File: generator/templates/cli/retrieveflag.gotmpl [hunk 3]
```diff
@@ -36,4 +47,34 @@
...
+
+        {{ $flagValueVar }} := make({{ .GoType }}, len({{ $flagValueVar }}Str))
+        for i, v := range {{ $flagValueVar }}Str {
+            if err := {{ $flagValueVar }}[i].UnmarshalText([]byte(v)); err != nil{
+                return err, false
+            }
+        }
+        m.{{ pascalize .Name }} = {{- if .IsNullable }}&{{- end }}{{ $flagValueVar }}
+    {{- else }}
+        // warning: primitive {{.Name}} {{.GoType }} is not supported by go-swagger cli yet
```
Comment (youyuanwu) [inline generator/templates/cli/retrieveflag.gotmpl:78]: primitive -> array

File: generator/templates/cli/retrieveflag.gotmpl (hunk 2 above)
Comment (dimovnike) [reply generator/templates/cli/retrieveflag.gotmpl:36]: They all use different form for initialization. For example you can not init UUID with {{ .GoType }}{} and ObjectId fails when initialized with empty string. However, if we use:

> var {{ $flagValueVar }} {{ .GoType }}

The same code works for all 3 cases. I will update the code soon.

File: generator/templates/cli/registerflag.gotmpl (hunk 1 above)
Comment (dimovnike) [reply generator/templates/cli/registerflag.gotmpl:75]: The suggested fix didn't work.

enumcompletion is broken for array items even for the primitive types (the code in master is not working). I suggest leaving this for another pull request.

File: generator/templates/cli/retrieveflag.gotmpl (hunk 2 above)
Comment (dimovnike) [reply generator/templates/cli/retrieveflag.gotmpl]: fixed as described.

File: generator/templates/cli/retrieveflag.gotmpl (hunk 3 above)
Comment (dimovnike) [reply generator/templates/cli/retrieveflag.gotmpl]: fixed.

File: generator/templates/cli/registerflag.gotmpl (hunk 1 above)
Comment (youyuanwu) [reply generator/templates/cli/registerflag.gotmpl:75]: You should then remove {{ template "enumcompletion" . }} in those places

**Review by youyuanwu — APPROVED** [review]

File: generator/templates/cli/registerflag.gotmpl (hunk 1 above)
Comment (dimovnike) [reply generator/templates/cli/registerflag.gotmpl:75]: > You should then remove {{ template "enumcompletion" . }} in those places

Done.

File: generator/bindata.go [hunk 4]
```diff
@@ -1,66 +1,66 @@
 // Code generated by go-bindata. DO NOT EDIT.
 // sources:
-// templates/cli/cli.gotmpl (8.129kB)
```
Comment (youyuanwu) [inline generator/bindata.go:3]: Are you using "github.com/kevinburke/go-bindata/go-bindata" ? 
The files are off.

Comment (youyuanwu) [reply generator/bindata.go:3]: @casualjim Do we have a bindata gen check in CI to ensure that data are correctly generated?

Comment (dimovnike) [reply generator/bindata.go:3]: I used github.com/shuLhan/go-bindata/. I will regenerate with github.com/kevinburke/go-bindata/go-bindata and will let you know.

Comment (dimovnike) [reply generator/bindata.go]: Done. Now the files look more similar to what's in master.

**Pull request merged**